packages/flutter_tools/lib/src/android/deferred_components_gen_snapshot_validator.dart

```diff
@@ -120,14 +127,16 @@ class DeferredComponentsGenSnapshotValidator extends DeferredComponentsValidator
     // Check for existing metadata entry and see if needs changes.
     bool exists = false;
     bool modified = false;
-    for (final XmlElement metaData in document.findAllElements('meta-data')) {
-      final String name = metaData.getAttribute('android:name');
-      if (name == _mappingKey) {
-        exists = true;
-        final String storedMappingString = metaData.getAttribute('android:value');
-        if (storedMappingString != encodedMapping) {
-          metaData.setAttribute('android:value', encodedMapping);
-          modified = true;
+    for (final XmlElement application in document.findAllElements('application')) {
```

Comment:
This ensures that it only looks for metadata element under 'application', not any other element.